### PR TITLE
148-update-dg-implementation-details

### PR DESCRIPTION
**Pull request by shayaansultan**
Update Developer Guide with implementation details for the TaskList and other features

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -162,6 +162,32 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+Step 4. The `addEmployee` adds the `Employee` to the `Model`.
+
+The following class diagram shows the structure of an `Employee`:
+
+<puml src="diagrams/UpdatedEmployeeClassDiagram.puml" width="450" />
+
+And this is the sequence diagram that describes the steps:
+
+<puml src="diagrams/DeleteSequenceDiagram.puml" width="550" />
```
Comment (hjungwoo01) [inline docs/DeveloperGuide.md:185]: Is this supposed to be the DeleteSequenceDiagram?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -236,32 +244,126 @@ Step 2. The `DeleteCommand` is executed, calling `Model#deleteEmployee(target)`,
 - **Why this design:** The command pattern is used for consistency with other commands in the application and to keep the parsing and execution logic separated. The use of target uid for deletion allows for precise and flexible removal of employees without hard-coding specific deletion types.
 - **Alternatives considered:** A direct approach where the `DeleteCommand` directly manipulates the employee list was considered but rejected to maintain a clean separation between the command and the model, adhering to the Single Responsibility Principle.
 
-### Add Command Implementation
+### Filter Command Implementation
 
-The `AddCommand` implements the abstract class `Command` to add a new `Employee` to the `Model`.
+The `FilterCommand` is implemented to allow users to refine the list of employees displayed based on specified criteria, such as role, team, and tags. This functionality is crucial for users who need to work with subsets of large employee datasets.
```
Comment (hjungwoo01) [inline docs/DeveloperGuide.md:249]: We need to add name here as well.

**Review by hjungwoo01 — CHANGES_REQUESTED** [review]
Most of it looks good, but I think some parts need a closer look.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (shayaansultan) [reply docs/DeveloperGuide.md]: Replaced with AddSequenceDiagram in new commit

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (shayaansultan) [reply docs/DeveloperGuide.md]: Name added in new commit

**Review by hjungwoo01 — APPROVED** [review]
LGTM!

**Pull request merged**